wandb/sdk/internal/datastore.py

```diff
@@ -231,8 +259,10 @@ def _write_data(self, s):
                 data_used += LEVELDBLOG_DATA_LEN
                 data_left -= LEVELDBLOG_DATA_LEN
 
-            # write last
+            # write last and flush the entire block to disk
             self._write_record(s[data_used:], LEVELDBLOG_LAST)
+            self._fp.flush()
+            os.fsync(self._fp.fileno())
```

Comment:
@raubitsj not positive this is the right thing to do, but figured we should at least flush every block to disk explicitly.  I don't think this actually fixes any issues with syncing and I can take it out if you think it will have negative performance consequences.

Reply:
This isnt quite doing what you think it is doing.  
I had planned on making flushes happen time based that is why i left it out.  Writing on every block is fine, but this is only syncing on a record that spanned multiple blocks.   It will be fine, but it wont actually guarantee that things are flushed regularly -- but in most cases it will happen.
lets keep it for now if it was tested to work ok.